src/ParallelAlgorithms/LinearSolver/Schwarz/Actions/CommunicateOverlapFields.hpp

```diff
@@ -124,6 +127,17 @@ struct SendOverlapFields<tmpl::list<OverlapFields...>, OptionsGroup,
         expand_pack((get<OverlapFields>(overlap_fields) =
                          db::get<OverlapFields>(box))...);
       }
+      // We elide the tagged tuple in the inbox tag if only a single tag is
+      // communicated. This optimization allows moving the overlap-map into the
+      // DataBox in one piece.
+      auto& collapsed_overlap_fields = [&overlap_fields]() noexcept -> auto& {
```

Comment:
is the `-> auto&` necessary for this lambda? I'm not sure of the rules here, but would have anticipated that it would infer a reference.

Reply:
My understand is that it is necessary, since the implicit return type of a lambda is always `auto`, i.e. return-by-value. When I remove the trailing return type the compiler warns me that I'm referencing a temporary.

Reply:
:+1: good to know